pkg/cluster/internal/providers/docker/network.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// TODO: we'll probably allow configuring this
+//
+// however currently picking a single network is equivilant to the previous
+// behavior *except* that we moved from the default bridge to a user defined
+// network because the default bridge is actually special versus any other
+// docker network and lacks the emebdded DNS
+//
+// for now this also makes it easier for apps to join the same network
```

Comment:
I.E. you can do `docker run --network=kind --name=metallb-test-cluster-sidecar ....` when doing fun test stuff.
Having a predictable default network makes that saner.

We could do one per cluster but not only is that more complex, it's going to cause IP allocation issues more readily when you have lots of small clusters (which is much moreso the usage pattern than one cluster with a large number of nodes).

In the future this can be configurable and this can just be the default, but it's roughly equivilant to the current state and keeps the PR focused on the necessary changes.

Reply:
also it will modify current behaviour where multiple cluster are in the same network.
IIRC istio was using it